hydromt/components/config.py

```diff
@@ -70,7 +71,7 @@ def _initialize(self, skip_read=False) -> None:
         """Initialize the model config."""
         if self._data is None:
             self._data = dict()
-            if self._root.is_reading_mode() and not skip_read:
+            if not skip_read:
```

Comment:
Can you re-add the sefl._root.is_reading_mode? I added it again because we are changing behavior and enforcing a create method in write mode for config.